systemservices/workflow/workflow.go

```diff
@@ -0,0 +1,66 @@
+package workflow
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/mesg-foundation/core/execution"
+	"github.com/mesg-foundation/core/utils/workflowparser"
+)
+
+// WSS's tasks.
+const (
+	CreateTaskKey = "create"
+	DeleteTaskKey = "delete"
+)
+
+// CreateInputs maps create task's inputs.
+// name is optional and has to be unique.
+func CreateInputs(definition workflowparser.WorkflowDefinition, name string) (map[string]interface{}, error) {
+	// TODO: this hack is not something that we should do but
+	// it's needed because *parameterValidator is not able to identify
+	// structs for now.
+	definitionData, err := json.Marshal(definition)
+	if err != nil {
+		return nil, err
+	}
+	var data interface{}
+	if err := json.Unmarshal(definitionData, &data); err != nil {
+		return nil, err
+	}
+	return map[string]interface{}{
+		"definition": data,
+		"name":       name,
+	}, nil
+}
+
+// CreateOutputs maps create task's outputs.
+// id is unique workflow id.
+func CreateOutputs(e *execution.Execution) (id string, err error) {
+	switch e.OutputKey {
+	case "success":
+		return e.OutputData["id"].(string), nil
+	case "error":
+		return "", errors.New(e.OutputData["message"].(string))
+	}
+	panic("unreachable")
```

Comment:
Could we get rid of panic?
Ok, the code should not be unreachable, but for a magical reason that get it reached, then it make the core crash.
Can we just return an error with a nice message?

Reply:
To me, this is where having tests comes in because they're meant to catch programmer errors. 

As you also said, panic should never happen. Our business logic does not depend on having one. If we get a panic anyway, it'd be due to a programmer error. Which means, we have a bug in Core or WSS. We should write tests to avoid buggy code to be merged which could cause to this panic instead of returning an error that will never get handled.

If you still think that we should return an error instead, I'm ok with it.

Reply:
In Discord, we agreed on using errors instead for now. In future we may change to goroutines again and recover any panic on gRPC handlers to prevent Core to crash.

Reply:
done